src/main/java/com/dragonballs/services/user/UserService.java

```diff
@@ -0,0 +1,49 @@
+package com.dragonballs.services.user;
+
+import com.dragonballs.dao.UserDAO;
+import com.dragonballs.entities.User;
+import com.dragonballs.exceptions.UserException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class UserService {
+    @Autowired
+    private UserDAO userDAO;
+
+    private BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    private UserValidator userValidator;
+
+    public UserService() {
+        bCryptPasswordEncoder = new BCryptPasswordEncoder();
+        userValidator = new UserValidator();
+    }
+
+    public User registerUser(User user) {
+        userValidator.validate(user);
+
+        User existingUser = userDAO.findByEmail(user.getEmail());
+
+        if (existingUser != null) {
+            throw new UserException("User already exists");
```

Comment:
To return 400 BAD  REQUEST sounds................ Not right

https://stackoverflow.com/a/9270432